github/repos.go

```diff
@@ -571,25 +598,58 @@ func (s *RepositoriesService) GetBranch(owner, repo, branch string) (*Branch, *R
 	return b, resp, err
 }
 
-// EditBranch edits the branch (currently only Branch Protection)
+// GetBranchProtection gets the protection of a given branch.
 //
-// GitHub API docs: https://developer.github.com/v3/repos/#enabling-and-disabling-branch-protection
-func (s *RepositoriesService) EditBranch(owner, repo, branchName string, branch *Branch) (*Branch, *Response, error) {
-	u := fmt.Sprintf("repos/%v/%v/branches/%v", owner, repo, branchName)
-	req, err := s.client.NewRequest("PATCH", u, branch)
+// GitHub API docs: https://developer.github.com/v3/repos/branches/#get-branch-protection
+func (s *RepositoriesService) GetBranchProtection(owner, repo, branch string) (*Protection, *Response, error) {
+	u := fmt.Sprintf("repos/%v/%v/branches/%v/protection", owner, repo, branch)
+	req, err := s.client.NewRequest("GET", u, nil)
 	if err != nil {
 		return nil, nil, err
 	}
+	req.Header.Set("Accept", mediaTypeProtectedBranchesPreview)
```

Comment:
Minor, but I think it'd be best to stick to the usual formula here for consistency. Add a blank line before `req.Header.Set` and this comment:

```
// TODO: remove custom Accept header when this API fully launches.
```

Just to keep this endpoint code looking more [like others](https://github.com/google/go-github/blob/7baedea38fd13aae4e4e9cb0854250797569a91c/github/git_commits.go#L67-L69).

Here, and in 2 other places.